refactor(wallet)!: Simplify public_descriptor(), remove redundant function

gnapoli23 opened the pull request:
Fixes #1501 


### Description

Simplifies `public_descriptor` function by using `get_descriptor` and removes `get_descriptor_for_keychain`.

### Notes to the reviewers

Tested with `cargo test --all-features`.

### Changelog notice

- Simplifies `public_descriptor` function and removes `get_descriptor_for_keychain`

### Checklists

#### All Submissions:

* [X] I've signed all my commits
* [X] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [X] I ran `cargo fmt` and `cargo clippy` before committing

notmandatory commented on crates/wallet/src/wallet/mod.rs:
```diff
@@ -1763,16 +1763,15 @@ impl Wallet {
+    /// It's the "public" version of the wallet's descriptor, meaning a new descriptor that has
     /// the same structure but with every secret key removed
     ///
     /// This can be used to build a watch-only version of a wallet
```
⛏️ Docs should be proper sentences ending with a period and now is a good time to clarify that public keys correspond to the descriptors secret keys. 
```suggestion
    /// the same structure but with the all secret keys replaced by their corresponding public key.
    ///
    /// This can be used to build a watch-only version of a wallet.
```

gnapoli23 replied:
Thanks for the suggestion, just commited.

notmandatory reviewed (changes requested):
Looks good, one suggestion for the docs.

notmandatory commented:
One final thing, you need to setup a github signing key and squash and sign your commit.
see: https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification

gnapoli23 commented:
> One final thing, you need to setup a github signing key and squash and sign your commit. see: https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification

Sure, I thought I signed it with my yubikey but I just pushed it without signing. Going to fix in a brief.

gnapoli23 commented:
Done. @notmandatory

notmandatory commented:
I just realized you also should update the commit message to conform to https://www.conventionalcommits.org/en/v1.0.0/. 

We don't usually put github issue numbers in commit messages, so something like this:

```
refactor(wallet)!: remove redundant Wallet::get_descriptor_for_keychain()

Simplify Wallet::public_descriptor() and update Wallet internals to use public_descriptor() instead of get_descriptor_for_keychain().
```

gnapoli23 commented:
@notmandatory Updated following your suggestions, please let me know if something else needs to be fixed. Thanks.

storopoli commented:
The commit message is still too wide, 50 chars for title and 72 for body:

Suggestion:

```
refactor(wallet)!: remove redundant get_descriptor_for_keychain

Simplify Wallet::public_descriptor() and update Wallet internals to use
public_descriptor() instead of get_descriptor_for_keychain().
```

gnapoli23 committed:
refactor(wallet)!: remove redundant get_descriptor_for_keychain

Simplify Wallet::public_descriptor() and update Wallet internals to use
public_descriptor() instead of get_descriptor_for_keychain().

gnapoli23 commented:
Ok, please check now. @notmandatory @storopoli

notmandatory reviewed (approved):
ACK e7ec5a873328c9b68836306086bd9b021e9c952a

Thanks for getting this one wrapped up.

storopoli reviewed (approved):
ACK https://github.com/bitcoindevkit/bdk/commit/e7ec5a873328c9b68836306086bd9b021e9c952a

gnapoli23 commented:
Thanks to both of you @notmandatory @storopoli . Can it be merged now?
